pkg/plugin/sdk/deployment.go

```diff
@@ -841,3 +859,73 @@ func (k ArtifactKind) toModelEnum() model.ArtifactVersion_Kind {
 		return model.ArtifactVersion_UNKNOWN
 	}
 }
+
+// DetermineStrategyInput is the input for the DetermineStrategy method.
+type DetermineStrategyInput struct {
+	// Request is the request to determine the strategy.
+	Request DetermineStrategyRequest
+	// Client is the client to interact with the piped.
+	Client *Client
+	// Logger is the logger to log the events.
+	Logger *zap.Logger
+}
+
+// DetermineStrategyRequest is the request to determine the strategy.
+type DetermineStrategyRequest struct {
+	// Deployment is the deployment that the strategy will be determined.
+	Deployment Deployment
+	// DeploymentSource is the source of the deployment.
+	DeploymentSource DeploymentSource
```

Comment:
We need runnningDeploySource and targetDeploySource according to the current implementation.
https://github.com/pipe-cd/pipecd/blob/d50a6765488ef5fe037c1314790b8bb39123663f/pkg/app/pipedv1/plugin/kubernetes/deployment/server.go#L82-L107

Reply:
Thank you.
Fixed on https://github.com/pipe-cd/pipecd/pull/5655/commits/024b273a89ca799a7349e9075b3051b01920745c